tas-client/src/tas-client/FeatureProvider/TasApiFeatureProvider.ts

```diff
@@ -38,7 +38,48 @@ export class TasApiFeatureProvider extends FilteredFeatureProvider {
         }
 
         //axios webservice call.
-        let response: AxiosResponse<TASFeatureData> = await this.httpClient.get({ headers: headers });
+        let response: AxiosResponse<TASFeatureData> | undefined;
+
+        
+        try {
+            /**
+             * As mentioned in the axios docs:
+             * Axios Promise will handle handle the catch
+             * section of promises whenever a request is not succesful.
+             * https://axios-http.com/docs/handling_errors
+             */
+            response = await this.httpClient.get({ headers: headers });
+        } catch(error) {
+            const axiosError = error as AxiosError;
+            const properties: Map<string, string> = new Map();
+            if (axiosError.response) {
+                // The request was made and the server responded with a status code
+                // that falls out of the range of 2xx
+                properties.set("RESPONSE_DATA", axiosError.response.data);
+                properties.set("RESPONSE_STATUS", `${axiosError.response.status}`);
+                properties.set("RESPONSE_HEADERS", `${axiosError.response.headers}`);
+            } else if (axiosError.request) {
+                // The request was made but no response was received
+                // `error.request` is an instance of XMLHttpRequest in the browser and an instance of
+                // http.ClientRequest in node.js
+                properties.set("ERROR_REQUEST", JSON.stringify(axiosError.request));
+            } else {
+                // Something happened in setting up the request that triggered an Error
+                properties.set("ERROR_MESSAGE", axiosError.message);
+            }
+            properties.set("ERROR_CONFIG", JSON.stringify(axiosError.config));
+            this.telemetry.postEvent("tasApiFetchError", properties);
+        }
+
+        // In case the response fetchin failed, return 
+        // default feature data.
+        if (!response){
+            return {
```

Comment:
This might be different behavior from what we have today.. This will overwrite whatever exists in the cache with no experiment info. In the past if the request failed, I think we might have just kept the cache as it was from the last successful fetch?

I'm not sure what the "correct" behavior is... Do you have thoughts?

Reply:
I think it will be best to maybe rethrow the exception just to keep this behaving as it was.
And make sure it's not disruptive on the data we currently have. I'll take a look at the code tomorrow and see what's the current situation (uncaught exception) vs returning empty cache response.

Reply:
I ended up throwing an error, that way we have the same flow as before.